novawallet/Common/Helpers/MetaAccountModel+Delegated.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+
+extension MetaAccountModel {
+    func isProxied(accountId: AccountId, chainId: ChainModel.Id) -> Bool {
+        type == .proxied && has(accountId: accountId, chainId: chainId)
+    }
+
+    func isSignatory(for multisig: MultisigAccountType) -> Bool {
+        switch multisig {
+        case let .universal(multisig):
+            multisig.signatory == substrateAccountId || multisig.signatory == ethereumAddress
+        case let .singleChain(chainAccount, multisig):
+            chainAccounts.contains {
+                $0.chainId == chainAccount.chainId && $0.accountId == multisig.signatory
+            }
+        }
+    }
+
+    func isDelegated() -> Bool {
+        type == .proxied || type == .multisig
+    }
+
+    func proxyChainAccount(
+        chainId: ChainModel.Id
+    ) -> ChainAccountModel? {
+        chainAccounts.first { $0.chainId == chainId && $0.proxy != nil }
+    }
+
+    func multisigAccount() -> MultisigAccountType? {
+        if let multisig {
+            .universal(multisig: multisig)
+        } else if let chainAccount = chainAccounts.first(where: { $0.multisig != nil }),
+                  let multisig = chainAccount.multisig {
+            .singleChain(chainAccount: chainAccount, multisig: multisig)
+        } else {
+            nil
+        }
+    }
+
+    func proxy() -> DelegatedAccount.ProxyAccountModel? {
+        guard type == .proxied,
+              let chainAccount = chainAccounts.first(where: { $0.proxy != nil }) else {
+            return nil
+        }
+
+        return chainAccount.proxy
+    }
+
+    func address(for chainAsset: ChainAsset) throws -> AccountAddress? {
+        let request = chainAsset.chain.accountRequest()
+        return fetch(for: request)?.toAddress()
+    }
+
+    func delegationId() -> MetaAccountDelegationId? {
+        switch type {
+        case .multisig:
+            guard
+                let (chainAccount, multisigModel) = multisigAccount()?.multisig,
+                let multisigModel
+            else { return nil }
+
+            return MetaAccountDelegationId(
+                delegatedAccountId: multisigModel.accountId,
+                delegatorId: multisigModel.signatory,
+                chainId: chainAccount?.chainId,
+                delegationType: .multisig
+            )
+        case .proxied:
+            guard
+                let proxyAccountId = proxy()?.accountId,
+                let proxiedAccount = chainAccounts.first(where: { $0.proxy?.accountId == proxyAccountId }),
+                let proxy = proxiedAccount.proxy
+            else { return nil }
+
+            return MetaAccountDelegationId(
+                delegatedAccountId: proxyAccountId,
```

Comment:
We have discussed that we will have `delegate` and `delegator` notion instead of delegated